pyls/workspace.py

```diff
@@ -29,6 +81,17 @@ def __init__(self, root_uri, lang_server=None):
         self._docs = {}
         self._lang_server = lang_server
 
+        # Whilst incubating, keep private
+        self.__rope = Project(self._root_path)
+        self.__rope.prefs.set('extension_modules', self.PRELOADED_MODULES)
+        # jedi.api.preload_module(*self.PRELOADED_MODULES)
+
+    @property
+    def _rope(self):
+        # TODO: we could keep track of dirty files and validate only those
+        self.__rope.validate()
```

Comment:
Does this actually work? I never tested it?

Reply:
I will take a look at it, however it's not generating any error

Reply:
The problems I ran into were performing a workspace rename, then undoing, then trying to do it again. Haven't tested this new implementation yet though!